tsl/src/compression/create.c

```diff
@@ -282,21 +288,92 @@ build_columndefs(CompressionSettings *settings, Oid src_relid)
 														   attr->atttypmod,
 														   attr->attcollation));
 		}
-		else if (bms_is_member(attr->attnum, btree_columns))
+		else if (bms_is_member(attr->attnum, index_columns))
 		{
-			TypeCacheEntry *type = lookup_type_cache(attr->atttypid, TYPECACHE_LT_OPR);
+			TypeCacheEntry *type =
+				lookup_type_cache(attr->atttypid, TYPECACHE_LT_OPR | TYPECACHE_HASH_EXTENDED_PROC);
+
+			/*
+			 * We can have various unusual user-defined types which do not
+			 * support comparison or hashing. The sparse indexes for the
+			 * non-orderby columns are not required for correctness, so just
+			 * don't create the sparse index if we lack the suitable operators.
+			 */
+			bool can_use_minmax = OidIsValid(type->lt_opr);
+
+			/*
+			 * For bloom1 indexes we currently use our custom hash functions
+			 * which have better characteristics.
+			 */
+			bool can_use_bloom1 = bloom1_get_hash_function(attr->atttypid) != NULL;
```

Comment:
Minor nit: next ~40 lines are all about determining if we can use bloom filter on the column. Might make sense to separate it out in a function for readability sake.